toolchain/driver/driver.cpp

```diff
@@ -30,6 +30,45 @@
 
 namespace Carbon {
 
+auto Driver::FindPreludeFiles(llvm::StringRef data_dir,
+                              llvm::raw_ostream& error_stream)
+    -> llvm::SmallVector<std::string> {
+  llvm::SmallVector<std::string> result;
+
+  // Include <data>/core/prelude.carbon, which is the entry point into the
+  // prelude.
+  {
+    llvm::SmallString<256> prelude_file(data_dir);
+    llvm::sys::path::append(prelude_file, llvm::sys::path::Style::posix,
+                            "core/prelude.carbon");
+    result.push_back(prelude_file.str().str());
+  }
+
+  // Glob for <data>/core/prelude/**/*.carbon and add all the files we find.
+  llvm::SmallString<256> prelude_dir(data_dir);
+  llvm::sys::path::append(prelude_dir, llvm::sys::path::Style::posix,
+                          "core/prelude");
+  std::error_code ec;
+  for (llvm::sys::fs::recursive_directory_iterator prelude_files_it(
+           prelude_dir, ec, /*follow_symlinks=*/false);
+       prelude_files_it != llvm::sys::fs::recursive_directory_iterator();
+       prelude_files_it.increment(ec)) {
+    if (ec) {
+      error_stream << "ERROR: Could not find prelude: " << ec.message() << "\n";
+      result.clear();
+      break;
```

Comment:
```suggestion
      return {};
```

I think it'd be clearer here to return directly.

Reply:
Sure. But let's not disable NRVO:
```suggestion
      result.clear();
      return result;
```